lib/src/viewport/transitions/fly_viewport_transition.dart

```diff
@@ -0,0 +1,30 @@
+part of mapbox_maps_flutter;
+
+/// A [ViewportTransition] that performs a fly animation over a specified duration.
+///
+/// The [FlyViewportTransition] animates the camera using a flight path that typically follows a zoom-out,
+/// flight, and zoom-in pattern. This creates a dynamic and engaging transition between different viewport
+/// states, giving users a sense of movement across the map.
+///
+/// ### Example
+///
+/// ```dart
+/// final flyTransition = FlyViewportTransition(
+///   duration: Duration(seconds: 5),
+/// );
+/// ```
+///
+/// In this example, the fly animation will occur over 5 seconds. If [duration] is not specified,
+/// the transition will use a default duration determined by the system.
+final class FlyViewportTransition extends ViewportTransition {
+  /// The duration of the fly animation.
+  ///
+  /// Specifies how long the fly animation should take to complete. If not provided, a default duration
+  /// will be used by the system.
```

Comment:
Which system here? Is this platform specific?

Reply:
If not specified the duration is calculated by the fly to interpolation algorithm, should be the same on both platform as the algorithm is the same.